### ISSUE TITLE
Orientation arrows in lower left corner would be more useful as part of a viewcube

### ISSUE THREAD

**Issue by d235j** [solvespace/solvespace#70]
The orientation arrows in the lower left corner currently seem to only be useful to provide an indication of the view. It would be nice if they could be extended to make them usable for navigation, as part of a viewcube allowing dragging around and selection of faces/edges.

**Comment by whitequark** [comment]
@jwesthues thoughts on this?

**Comment by jwesthues** [comment]
I guess @d235j doesn't have a center mouse button? I could see the utility if not, and there's some vague analogy with e.g. dragging imported part normals to rotate the part.

**Comment by Symbian9** [comment]
Think, @d235j expect something like Blender manipulator, yes?
![Blender 3D manipuator](https://www.blender.org/manual/ru/_images/editors_3dview_transform_control-manipulators-manipulator_options.png)

* https://www.blender.org/manual/ru/editors/3dview/transform/transform_control/manipulators.html

**Comment by d235j** [comment]
Yeah, exactly like the Blender manipulator. There are a couple of reasons for this, one being ease of use on devices without a center mouse button, but more importantly it would make it much easier to jump to a specific view without having to hunt for menu options.

**Comment by Symbian9** [comment]
@d235j, I only can't understand what you mean in this:
> and selection of faces/edges.

Can you give some pictures?

**Comment by whitequark** [comment]
> I guess @d235j doesn't have a center mouse button?

Center *and* right mouse button; I have a laptop with only left and right buttons, and Shift+RMB is fairly usable (though awkward at first), but I imagine it's really painful on an Apple trackpad. Gestures (https://github.com/solvespace/solvespace/issues/68) would also help here.

**Comment by Evil-Spirit** [comment]
Viewcube is for selecing projections/views. We click on vertices/edges/faces to align view.
![2016-10-12 21 38 28](https://cloud.githubusercontent.com/assets/1980537/19314455/5ea396fc-90c4-11e6-95f0-387fe6b63923.png)

**Comment by jwesthues** [comment]
The x-y-z normals in the corner can also be constrained against, so the viewcube alone would lose functionality. I guess you could do a modified viewcube that also kept that. For iso/ortho views, it's faster than the menus, but the keyboard shortcuts are faster still. SolveSpace's interface would require considerable changes beyond the viewcube to be well-suited to touchscreen devices.

> Center and right mouse button; I have a laptop with only left and right buttons, and Shift+RMB is fairly usable (though awkward at first)

I guess we could add more keyboard modifiers, to permit all three operations (select, translate, rotate) with left.

**Comment by d235j** [comment]
I think a modified viewcube would work well.
Keyboard shortcuts, while great for an experienced user, mean an increased learning curve and are not very intuitive for someone starting out.

**Comment by Symbian9** [comment]
> I guess we could add more keyboard modifiers, to permit all three operations (select, translate, rotate) with left.

@jwesthues, it would be better make fully configurated keyboard shortcuts that any user can set by himself or set default.
